src/typesystem.rs

```diff
@@ -115,7 +115,10 @@ pub enum DataTypeInformation {
         size: u32,
     },
     String {
-        size: u32,
+        ///the size-expression used in the declaration (e.g. STRING(`LEN`))
+        size: Option<ConstId>,
+        ///size to be used if the parameter `size` is None
+        default_size: u32,
```

Comment:
This seems strange to me. I would rather we treat STRING and WSTRING as aliases to STRING[80] and WSTRING[80] or 81

Reply:
After looking a bit more into how this is used, Iff we can't treat strings just like the arrays, I think we should solve this using a single type / enum  : 
```rust
enum Size {
  Literal(u32),
  Expression(ConstId),
}
```
This would ensure we always handle both cases when we encounter them.
I assume such a "Size" enum would also work well on Array Dimensions and subranges in that case.

Reply:
done - implemented as you suggested - I called it TypeSize